docs/cli-usage.md

```diff
@@ -0,0 +1,141 @@
+below.  Same list can be provided via INSTANCES environment variable, in case
+when both are provided - proxy will use command line flag.
+
+* `-instances_metadata=metadata_key`
+
+Usable on [GCE](https://cloud.google.com/compute/docs/quickstart) only. The
+given [GCE metadata](https://cloud.google.com/compute/docs/metadata) key will be
+polled for a list of instances to open in `-dir`. The metadata key is relative
+from `computeMetadata/v1/`. The format for the value is the same as the
+'instances' flag. A hanging-poll strategy is used, meaning that changes to the
+metadata value will be reflected in the `-dir` even while the proxy is running.
+When an instance is removed from the list the corresponding socket will be
+removed from `-dir` as well (unless it was also specified in `-instances`), but
+any existing connections to this instance will NOT be terminated.
+
+Note: `-instances` and `-instances_metadata` may be used at the same time but
+are not compatible with the `-fuse` flag.
+
+* `-ip_address_types=PUBLIC,PRIVATE`
+
+A comma-delimited list of preferred IP types for connecting to an instance. For
+example, setting this to PRIVATE will force the proxy to connect to instances
+using an instance's associated private IP. Defaults to `PUBLIC,PRIVATE`
+
+* `-term_timeout=30s`
+
+How long to wait for connections to close before shutting down the proxy.
+Defaults to 0.
+
+* `-skip_failed_instance_config`
+
+Setting this flag will allow you to prevent the proxy from terminating when some
+instance configurations could not be parsed and/or are unavailable.
+
+* `-log_debug_stdout=true`
+
+This is to log non-error output to stdOut instead of stdErr. For example, if you
+don't want connection related messages to log as errors, set this flag to true.
+Defaults to false.
+
+* `-credential_file`
+
+Allows use of the proxy outside of Google's cloud. Simply [create a new service
+account](https://cloud.google.com/sql/docs/mysql/sql-proxy#create-service-account),
+download the associated JSON file, and set `-credential_file` to the path of the
+JSON file. You can also set the GOOGLE_APPLICATION_CREDENTIALS environment
+variable instead of passing this flag.
+
+## Authentication
+
+Authentication can be configured in a few different ways. Those listed higher on
+the list will override options lower on the list:
+
+1. `credential_file` flag
+2. `token` flag
+3. Service account key at path stored in `GOOGLE_APPLICATION_CREDENTIALS`
+4. gcloud _user_ credentials (set from `gcloud auth login`)
+5. Default Application Credentials via goauth:
+
+   1. `GOOGLE_APPLICATION_CREDENTIALS` (again)
+   2. gcloud _application default_ credentials (set from ` gcloud auth application-default login`)
+   3. appengine.AccessToken (for App Engine Go < =1.9)
+   4. GCE/GAE metadata credentials
+
+Note: any account connecting to a Cloud SQL database will need the
+`roles/cloudsql.client` role. See [Roles and Permissions in Cloud
```

Comment:
Please specifically list the IAM permissions needed or`Cloud SQL client` role or higher. The App Engine page does a pretty good job:
![image](https://user-images.githubusercontent.com/31518063/108433088-42b73e00-71fa-11eb-9192-38f4e523883e.png)